warehouse/config/WHDefaultConfig.js

```diff
@@ -0,0 +1,22 @@
+const { coalesce } = require("./helpers");
```

Comment:
How about we reverse the config?

{
  "id": "messageId",  
  "user_id": "userId", 
  "context_request_ip": "request_ip", 
   ".." , 
  "context_ip": (msg) => msg .context.ip ||   msg.request_ip 
}

We check if the value is a function, If it is we fill it with the return value of the function. Makes it more readable.

Reply:
This seems easier to read. Basically, we have columns in DB to fill and we are filling this way.